### ISSUE TITLE
Fails to launch on EOS8

### ISSUE THREAD

**Issue by codecowboy** [pantheon-tweaks/pantheon-tweaks#259]
I am getting the following error when trying to launch pantheon tweaks on EOS8 (latest iso download as of Dec 24 2024). Launching from the applications menu simply fails silently. Launching from terminal gives:
```

flatpak run --verbose io.github.pantheon_tweaks.pantheon-tweaks
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/lukemackenzie/.local/share/flatpak
F: Opening user flatpak installation at path /home/lukemackenzie/.local/share/flatpak
F: Opening user flatpak installation at path /home/lukemackenzie/.local/share/flatpak
F: /home/xxxxxxxx/.local/share/flatpak/runtime/org.gnome.Platform/x86_64/47/5dbd45c448db6ab4a340bcd735ecc05cff6529ee6c0dd77ed47574c57508e13f/files/lib32 does not exist
F: Cleaning up unused container id 220918119
F: Allocated instance id 3739671345
F: Add defaults in dir /io/github/pantheon_tweaks/pantheon-tweaks/
F: Add locks in dir /io/github/pantheon_tweaks/pantheon-tweaks/
F: Disallowing network access
F: Allowing host-fs access
F: Not sharing "/run/media" with sandbox: Unable to open path "/run/media": No such file or directory
F: Allowing wayland access
F: Allowing x11 access
F: Running 'bwrap --args 41 -- xdg-dbus-proxy --args=43'
F: Running 'bwrap --args 41 -- start-pantheon-tweaks'
Failed to register: Timeout was reached
rmdir: failed to remove '/run/user/1000/dconf-bridge': Directory not empty
```

The directory indicated did not exist. I tried creating it which made no difference. There is a closed issue which is similar but the solution suggested there did not work

**Comment by ryonakano** [comment]
Hmm I can't reproduce your issue. Could you tell me your hardware information? It complains about timeout so I suppose you might be running on a slow device.

**Comment by filltr** [comment]
Same issue as OP.
Hardware:
12th Gen Intel® Core(TM) i7-1255U 
512 GB
32 GB memory
512 GB NVME

**Comment by ryonakano** [comment]
Thank you for your info!

@codecowboy Could you also tell me if you're using Secure Session (Wayland) or Classic Session (X11)?

**Comment by filltr** [comment]
Switching to secure session fix my problems.

**Comment by ryonakano** [comment]
@codecowboy @filltr Thank you two for testing. I'm actually using Classic Session and can't confirm this issue, so I supposed you're possibly using Secure Session. So, I'm not sure if which session a user using is relevant here thus I can't write that to README.

**Comment by ryonakano** [comment]
> So I would say it is still an issue which needs resolving.

Yes, I agree with you. I'd like to investigate this soon because it's critical that the app doesn't launch, but unfortunately, I haven't been able to reproduce this yet :dizzy_face:

I'd like to ask anyone who can confirm this issue:

- Does this issue only happen with Pantheon Tweaks? Is it possible to run other Flatpak apps like Music or Calcurator?
- How did you install Pantheon Tweaks? (by `flatpak install` command, or through Sideload app, etc.)

**Comment by ryonakano** [comment]
> Can you explain what `--device=dri` does?

This allows the specified Flatpak app to use the GPU acceleration.

From `man flatpak-override`:

```
       --device=DEVICE
           Expose a device to the application. This overrides to the Context section from the application metadata.  DEVICE must be one of: dri, kvm, shm, all. This option can be used
           multiple times.
```

and description about each DEVICE is available at https://docs.flatpak.org/en/latest/sandbox-permissions.html#device-access.

**Comment by ryonakano** [comment]
FYI the command is equivalent to turn on the `GPU Acceleration` setting in System Settings→Applications→Pantheon Tweaks:

![スクリーンショット 2024-12-30 23 02 00](https://github.com/user-attachments/assets/6c27bf06-9db6-4e40-bab5-a7a035b68da6)

**Comment by ryonakano** [comment]
> Did you already release a fixed version to flathub?

Not yet, I was going to release if it really fixes your issue.

As I couldn't confirm this issue until I tested on a real machine, it seems to depend on a hardware whether this issue happens or not.

Would it be possible for you to test on the same machine you encountered the issue?

**Comment by ryonakano** [comment]
@jwallat Great to hear that, thank you!

@codecowboy I'll go ahead and release the hotfix in a few days because this affects to some people. Let me know if that fix doesn't work for you when you reinstall your machine.

**Comment by ryonakano** [comment]
Thank you for your info.

> So maybe you are only supposed to set one of these and for elementaryOS it should be only `fallback-x11` which will use x11 only if there is no wayland socket?

I tried this but resulting a broken theme, so I don't think this is a solution:

![Screenshot From 2025-01-05 12-12-35](https://github.com/user-attachments/assets/ed1d62f7-ce9b-418a-9356-42d979b7933c)

However, changing the order may worth trying, while `--device=dri` might be still relevant because Calculator has it but Screenshot and File Roller don't have.

I've created a test branch https://github.com/flathub/io.github.pantheon_tweaks.pantheon-tweaks/pull/5 that just changed the order of sockets arguments. @codecowboy Could you try this on the original machine, with the following steps?

1. Reset the permission you gave to Pantheon Tweaks to the default from System Settings→Applications→Pantheon Tweaks→`Reset to Defaults` button
2. Uninstall current installation of Pantheon Tweaks completely with: `flatpak uninstall --delete-data -y io.github.pantheon_tweaks.pantheon-tweaks`
3. Install the test branch with: `flatpak install --user -y https://dl.flathub.org/build-repo/155185/io.github.pantheon_tweaks.pantheon-tweaks.flatpakref`
4. Reboot and login to Classic Session
5. Try launching Pantheon Tweaks

**Comment by ryonakano** [comment]
Thank you for trying it out. The order does not matter here then.

@codecowboy Could you try the following steps on the original machine?

1. Reset the permission you gave to Pantheon Tweaks to the default from System Settings→Applications→Pantheon Tweaks→`Reset to Defaults` button
2. Uninstall current installation of Pantheon Tweaks completely with: `flatpak uninstall --delete-data -y io.github.pantheon_tweaks.pantheon-tweaks`
3. Install the test branch with: `flatpak install --user -y https://dl.flathub.org/build-repo/155682/io.github.pantheon_tweaks.pantheon-tweaks.flatpakref`
4. Reboot and login to Classic Session
5. Try launching Pantheon Tweaks

Almost similar to the previous one but this time the branch to install is https://github.com/flathub/io.github.pantheon_tweaks.pantheon-tweaks/pull/4.

**Comment by ryonakano** [comment]
Released a hotfix https://github.com/flathub/io.github.pantheon_tweaks.pantheon-tweaks/pull/4, which should be available in a few days. Let me know if someone still faces the issue even after update.
